packages/components/src/components/Table/Table.js

```diff
@@ -0,0 +1,169 @@
+                        selectedRows,
+                        getBatchActionProps().onCancel
+                      );
+                    }}
+                  >
+                    {button.text}
+                  </TableBatchAction>
+                ))}
+              </TableBatchActions>
+              <TableToolbarContent>
+                {toolbarButtons.map(button => (
+                  <Button
+                    disabled={loading}
+                    onClick={button.handler}
+                    renderIcon={button.icon}
+                    key={`${button.text}Button`}
+                  >
+                    {button.text}
+                  </Button>
+                ))}
+              </TableToolbarContent>
+            </TableToolbar>
+            {loading ? (
+              <DataTableSkeleton
+                useZebraStyles
+                rowCount={1}
+                columnCount={headers.length}
+              />
+            ) : (
+              <CarbonTable>
+                <TableHead>
+                  <TableRow>
+                    {batchActionButtons.length > 0 && (
+                      <TableSelectAll {...getSelectionProps()} />
+                    )}
+                    {headers.map(header => {
+                      return (
+                        <TableHeader
+                          {...getHeaderProps({ header })}
+                          className="cellText"
+                          key={header.key}
+                        >
+                          {header.header}
+                        </TableHeader>
+                      );
+                    })}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {rows.map(row => {
+                    return (
+                      <TableRow {...getRowProps({ row })} key={row.id}>
+                        {batchActionButtons.length > 0 && (
+                          <TableSelectRow {...getSelectionProps({ row })} />
+                        )}
+                        {row.cells.map(cell => (
+                          <TableCell key={cell.id} id={cell.id}>
+                            {cell.value}
+                          </TableCell>
+                        ))}
+                      </TableRow>
+                    );
+                  })}
+                </TableBody>
+              </CarbonTable>
+            )}
+          </TableContainer>
+        )}
+      />
+      {!loading && dataRows.length === 0 && (
+        <div className="noRows">
+          {selectedNamespace === ALL_NAMESPACES
+            ? emptyTextAllNamespaces
+            : emptyTextSelectedNamespace}
+        </div>
+      )}
+    </div>
+  );
+};
+
+Table.propTypes = {
+  emptyTextAllNamespaces: PropTypes.string.isRequired,
+  emptyTextSelectedNamespace: PropTypes.string.isRequired,
+  headers: PropTypes.arrayOf(PropTypes.object).isRequired,
+  loading: PropTypes.bool.isRequired,
```

Comment:
Is this really required? If it's not provided it'll be `undefined` which is falsey, or if we wanted we could default to `false` in `defaultProps`.

`isRequired` is generally reserved for those props where we can't provide a sensible default.

There are also other props missing here.